PGVector fails to instantiate once micronaut-data-connection-jdbc library is on the classpath

### Expected Behavior

When I configure PG Vector integration then the `PgVectorEmbeddingStoreFactory` creates the `PgVectorEmbeddingStore` bean.

### Actual Behaviour

When `micronaut-data-connection-jdbc` library is on the classpath, it wraps every connection into `io.micronaut.data.connection.jdbc.advice.ContextualAwareDataSource.DataSourceProxy` but during the instantiation of `PgVectorEmbeddingStore` the ` io.micronaut.core.propagation.PropagatedContext` is not populated and _No current connection present_ exception is thrown.

```
Message: No current connection present. Consider declaring @Connectable or @Transactional on the surrounding method
Path Taken:
@i.m.c.a.EachBean d.l.s.e.p.PgVectorEmbeddingStore i.m.l.p.PgVectorEmbeddingStoreFactory.pgVectorEmbeddingStore#pgVectorEmbeddingStore(PgVectorEmbeddingStoreConfig configuration)
	at app//io.micronaut.context.DefaultBeanContext.resolveByBeanFactory(DefaultBeanContext.java:2357)
	at app//io.micronaut.context.DefaultBeanContext.createRegistration(DefaultBeanContext.java:3162)
	at app//io.micronaut.context.SingletonScope.getOrCreate(SingletonScope.java:80)
	at app//io.micronaut.context.DefaultBeanContext.intializeEagerBean(DefaultBeanContext.java:3051)
	at app//io.micronaut.context.DefaultBeanContext.initializeEagerBean(DefaultBeanContext.java:2711)
	at app//io.micronaut.context.DefaultBeanContext.initializeContext(DefaultBeanContext.java:2039)
	at app//io.micronaut.context.DefaultApplicationContext.initializeContext(DefaultApplicationContext.java:323)
	at app//io.micronaut.context.DefaultBeanContext.configureAndStartContext(DefaultBeanContext.java:3362)
	at app//io.micronaut.context.DefaultBeanContext.start(DefaultBeanContext.java:345)
	at app//io.micronaut.context.DefaultApplicationContext.start(DefaultApplicationContext.java:225)
	at app//io.micronaut.test.extensions.AbstractMicronautExtension.startApplicationContext(AbstractMicronautExtension.java:524)
	at app//io.micronaut.test.extensions.AbstractMicronautExtension.beforeClass(AbstractMicronautExtension.java:355)
	at app//io.micronaut.test.extensions.junit5.MicronautJunit5Extension.beforeAll(MicronautJunit5Extension.java:86)
	at java.base@21.0.9/java.util.ArrayList.forEach(ArrayList.java:1596)
Caused by: io.micronaut.data.connection.exceptions.NoConnectionException: No current connection present. Consider declaring @Connectable or @Transactional on the surrounding method
	at app//io.micronaut.data.connection.exceptions.NoConnectionException.notFoundInAdvice(NoConnectionException.java:49)
	at app//io.micronaut.data.connection.jdbc.advice.ContextualConnectionInterceptor.intercept(ContextualConnectionInterceptor.java:58)
	at app//io.micronaut.aop.chain.MethodInterceptorChain.proceed(MethodInterceptorChain.java:143)
	at app//io.micronaut.data.connection.jdbc.advice.ContextualConnection$Intercepted.createStatement(Unknown Source)
	at app//dev.langchain4j.store.embedding.pgvector.PgVectorEmbeddingStore.getConnection(PgVectorEmbeddingStore.java:414)
	at app//dev.langchain4j.store.embedding.pgvector.PgVectorEmbeddingStore.initTable(PgVectorEmbeddingStore.java:178)
	at app//dev.langchain4j.store.embedding.pgvector.PgVectorEmbeddingStore.<init>(PgVectorEmbeddingStore.java:95)
	at app//io.micronaut.langchain4j.pgvector.PgVectorEmbeddingStoreFactory$DataSourcePgVectorStore.<init>(PgVectorEmbeddingStoreFactory.java:50)
	at app//io.micronaut.langchain4j.pgvector.PgVectorEmbeddingStoreFactory.pgVectorEmbeddingStore(PgVectorEmbeddingStoreFactory.java:44)
	at app//io.micronaut.langchain4j.pgvector.$PgVectorEmbeddingStoreFactory$PgVectorEmbeddingStore0$Definition.instantiate(Unknown Source)
	at app//io.micronaut.context.BeanDefinitionDelegate.instantiate(BeanDefinitionDelegate.java:183)
	at app//io.micronaut.context.DefaultBeanContext.resolveByBeanFactory(DefaultBeanContext.java:2342)
	... 13 more
```

### Steps To Reproduce

1. check out this repository https://github.com/musketyr/jpa-langchain4j-conflict
2. run `JpaLangchain4jConflictTest`

### Environment Information


Operating system: macOS 26.1 (aarch64)
Java VM: Azul Zulu OpenJDK 64-Bit Server VM 21.0.8+9-LTS (mixed mode, emulated-client, sharing)	

### Example Application

https://github.com/musketyr/jpa-langchain4j-conflict

### Version

4.10.7

--- a/micronaut-langchain4j-store-pgvector/src/main/java/io/micronaut/langchain4j/pgvector/PgVectorEmbeddingStoreFactory.java
+++ b/micronaut-langchain4j-store-pgvector/src/main/java/io/micronaut/langchain4j/pgvector/PgVectorEmbeddingStoreFactory.java
@@ -1,5 +1,5 @@
 /*
- * Copyright 2017-2024 original authors
+ * Copyright 2017-2026 original authors
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -22,6 +22,9 @@ import io.micronaut.context.annotation.Bean;
 import io.micronaut.context.annotation.Context;
 import io.micronaut.context.annotation.EachBean;
 import io.micronaut.context.annotation.Factory;
+import io.micronaut.jdbc.DataSourceResolver;
+import org.jspecify.annotations.Nullable;
+
 import javax.sql.DataSource;
 
 /**
@@ -29,12 +32,22 @@ import javax.sql.DataSource;
  */
 @Factory
 public class PgVectorEmbeddingStoreFactory {
+    private final @Nullable DataSourceResolver dataSourceResolver;
+
+    public PgVectorEmbeddingStoreFactory(@Nullable DataSourceResolver dataSourceResolver) {
+        this.dataSourceResolver = dataSourceResolver;
+    }
+
     @EachBean(PgVectorEmbeddingStoreConfig.class)
     @Context
     @Bean(typed = EmbeddingStore.class)
     protected PgVectorEmbeddingStore pgVectorEmbeddingStore(PgVectorEmbeddingStoreConfig configuration) {
+        DataSource dataSource = configuration.getDataSource();
+        if (dataSourceResolver != null) {
+            dataSource = dataSourceResolver.resolve(dataSource);
+        }
         return new DataSourcePgVectorStore(
-                configuration.getDataSource(),
+                dataSource,
                 configuration.getTable(),
                 configuration.getDimension(),
                 configuration.getUseIndex(),

--- a/micronaut-langchain4j-store-pgvector/src/main/java/io/micronaut/langchain4j/pgvector/PgVectorEmbeddingStoreFactory.java
+++ b/micronaut-langchain4j-store-pgvector/src/main/java/io/micronaut/langchain4j/pgvector/PgVectorEmbeddingStoreFactory.java
@@ -23,6 +23,7 @@ import io.micronaut.context.annotation.Context;
 import io.micronaut.context.annotation.EachBean;
 import io.micronaut.context.annotation.Factory;
 import io.micronaut.jdbc.DataSourceResolver;
+import jakarta.inject.Inject;
 import org.jspecify.annotations.Nullable;
 
 import javax.sql.DataSource;
@@ -34,6 +35,11 @@ import javax.sql.DataSource;
 public class PgVectorEmbeddingStoreFactory {
     private final @Nullable DataSourceResolver dataSourceResolver;
 
+    public PgVectorEmbeddingStoreFactory() {
+        this(null);
+    }
+
+    @Inject
     public PgVectorEmbeddingStoreFactory(@Nullable DataSourceResolver dataSourceResolver) {
         this.dataSourceResolver = dataSourceResolver;
     }

--- a/micronaut-langchain4j-store-pgvector/src/test/java/io/micronaut/langchain4j/pgvector/PgVectorEmbeddingStoreFactoryTest.java
+++ b/micronaut-langchain4j-store-pgvector/src/test/java/io/micronaut/langchain4j/pgvector/PgVectorEmbeddingStoreFactoryTest.java
@@ -37,6 +37,20 @@ class PgVectorEmbeddingStoreFactoryTest {
         Assertions.assertNotNull(embeddingStore);
     }
 
+    @Test
+    void supportsConstructionWithoutDataSourceResolver() {
+        PgVectorEmbeddingStoreFactory factory = new PgVectorEmbeddingStoreFactory();
+        PgVectorEmbeddingStoreConfig configuration = new PgVectorEmbeddingStoreConfig(new StubDataSource(false), null);
+        configuration.setTable("test");
+        configuration.setDimension(384);
+        configuration.setCreateTable(false);
+        configuration.setDropTableFirst(false);
+
+        PgVectorEmbeddingStore embeddingStore = factory.pgVectorEmbeddingStore(configuration);
+
+        Assertions.assertNotNull(embeddingStore);
+    }
+
     private static final class StubDataSource implements DataSource {
         private final boolean failOnConnection;
 
@@ -57,7 +71,9 @@ class PgVectorEmbeddingStoreFactoryTest {
                                 Statement.class.getClassLoader(),
                                 new Class<?>[] {Statement.class},
                                 (statementProxy, statementMethod, statementArgs) -> switch (statementMethod.getName()) {
-                                    case "execute", "executeUpdate", "close" -> null;
+                                    case "execute" -> false;
+                                    case "executeUpdate" -> 0;
+                                    case "close" -> null;
                                     case "getConnection" -> proxy;
                                     case "unwrap" -> statementArgs[0] == Statement.class ? statementProxy : null;
                                     case "isWrapperFor" -> statementArgs[0] == Statement.class;

--- a/micronaut-langchain4j-store-pgvector/src/test/java/io/micronaut/langchain4j/pgvector/PgVectorEmbeddingStoreFactoryTest.java
+++ b/micronaut-langchain4j-store-pgvector/src/test/java/io/micronaut/langchain4j/pgvector/PgVectorEmbeddingStoreFactoryTest.java
@@ -84,10 +84,12 @@ class PgVectorEmbeddingStoreFactoryTest {
 
         @Override
         public void setLogWriter(PrintWriter out) {
+            // No-op: this test stub does not track mutable log-writer state.
         }
 
         @Override
         public void setLoginTimeout(int seconds) {
+            // No-op: this test stub does not enforce login timeouts.
         }
 
         @Override

--- a/micronaut-langchain4j-store-pgvector/src/test/java/io/micronaut/langchain4j/pgvector/PgVectorEmbeddingStoreFactoryTest.java
+++ b/micronaut-langchain4j-store-pgvector/src/test/java/io/micronaut/langchain4j/pgvector/PgVectorEmbeddingStoreFactoryTest.java
@@ -1,0 +1,147 @@
+package io.micronaut.langchain4j.pgvector;
+
+import dev.langchain4j.store.embedding.pgvector.PgVectorEmbeddingStore;
+import io.micronaut.jdbc.DataSourceResolver;
+import java.io.PrintWriter;
+import java.lang.reflect.Proxy;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.SQLFeatureNotSupportedException;
+import java.sql.Statement;
+import java.util.logging.Logger;
+import javax.sql.DataSource;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+class PgVectorEmbeddingStoreFactoryTest {
+
+    @Test
+    void unwrapsDataSourceBeforeConstructingStore() {
+        DataSource targetDataSource = new StubDataSource(false);
+        DataSource contextualDataSource = new StubDataSource(true);
+        DataSourceResolver dataSourceResolver = new DataSourceResolver() {
+            @Override
+            public DataSource resolve(DataSource dataSource) {
+                return targetDataSource;
+            }
+        };
+        PgVectorEmbeddingStoreFactory factory = new PgVectorEmbeddingStoreFactory(dataSourceResolver);
+        PgVectorEmbeddingStoreConfig configuration = new PgVectorEmbeddingStoreConfig(contextualDataSource, null);
+        configuration.setTable("test");
+        configuration.setDimension(384);
+        configuration.setCreateTable(false);
+        configuration.setDropTableFirst(false);
+
+        PgVectorEmbeddingStore embeddingStore = factory.pgVectorEmbeddingStore(configuration);
+
+        Assertions.assertNotNull(embeddingStore);
+    }
+
+    private static final class StubDataSource implements DataSource {
+        private final boolean failOnConnection;
+
+        private StubDataSource(boolean failOnConnection) {
+            this.failOnConnection = failOnConnection;
+        }
+
+        @Override
+        public Connection getConnection() throws SQLException {
+            if (failOnConnection) {
+                throw new SQLException("No current connection present");
+            }
+            return (Connection) Proxy.newProxyInstance(
+                    Connection.class.getClassLoader(),
+                    new Class<?>[] {Connection.class},
+                    (proxy, method, args) -> switch (method.getName()) {
+                        case "createStatement" -> Proxy.newProxyInstance(
+                                Statement.class.getClassLoader(),
+                                new Class<?>[] {Statement.class},
+                                (statementProxy, statementMethod, statementArgs) -> switch (statementMethod.getName()) {
+                                    case "execute", "executeUpdate", "close" -> null;
+                                    case "getConnection" -> proxy;
+                                    case "unwrap" -> statementArgs[0] == Statement.class ? statementProxy : null;
+                                    case "isWrapperFor" -> statementArgs[0] == Statement.class;
+                                    default -> defaultValue(statementMethod.getReturnType());
+                                }
+                        );
+                        case "close" -> null;
+                        case "unwrap" -> args[0] == Connection.class ? proxy : null;
+                        case "isWrapperFor" -> args[0] == Connection.class;
+                        default -> defaultValue(method.getReturnType());
+                    }
+            );
+        }
+
+        @Override
+        public Connection getConnection(String username, String password) throws SQLException {
+            return getConnection();
+        }
+
+        @Override
+        public PrintWriter getLogWriter() {
+            return null;
+        }
+
+        @Override
+        public void setLogWriter(PrintWriter out) {
+        }
+
+        @Override
+        public void setLoginTimeout(int seconds) {
+        }
+
+        @Override
+        public int getLoginTimeout() {
+            return 0;
+        }
+
+        @Override
+        public Logger getParentLogger() throws SQLFeatureNotSupportedException {
+            throw new SQLFeatureNotSupportedException();
+        }
+
+        @Override
+        public <T> T unwrap(Class<T> iface) throws SQLException {
+            if (iface.isInstance(this)) {
+                return iface.cast(this);
+            }
+            throw new SQLException("Unsupported unwrap target: " + iface.getName());
+        }
+
+        @Override
+        public boolean isWrapperFor(Class<?> iface) {
+            return iface.isInstance(this);
+        }
+    }
+
+    private static Object defaultValue(Class<?> returnType) {
+        if (!returnType.isPrimitive()) {
+            return null;
+        }
+        if (returnType == boolean.class) {
+            return false;
+        }
+        if (returnType == byte.class) {
+            return (byte) 0;
+        }
+        if (returnType == short.class) {
+            return (short) 0;
+        }
+        if (returnType == int.class) {
+            return 0;
+        }
+        if (returnType == long.class) {
+            return 0L;
+        }
+        if (returnType == float.class) {
+            return 0F;
+        }
+        if (returnType == double.class) {
+            return 0D;
+        }
+        if (returnType == char.class) {
+            return '\0';
+        }
+        return null;
+    }
+}

